File: app/src/main/java/com/edanuryildirim/futbolapi/view/DetailActivity.java
package com.edanuryildirim.futbolapi.view;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.edanuryildirim.futbolapi.R;
import com.edanuryildirim.futbolapi.adapter.DetailAdapter;
import com.edanuryildirim.futbolapi.model.FutbolModel;
import com.edanuryildirim.futbolapi.service.FutbolAPI;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class DetailActivity extends AppCompatActivity {

    ArrayList<FutbolModel> futbolModels;
    private String BASE_URL = "https://raw.githubusercontent.com/";
    Retrofit retrofit;
    RecyclerView recyclerView;
    DetailAdapter recyclerViewAdapter;
    TextView date;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        recyclerView = findViewById(R.id.recyclerViewDetail);
        date = findViewById(R.id.date);
        //Retrofit & JSON
        Gson gson = new GsonBuilder().setLenient().create();

        retrofit= new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build();

        loadDetailData();

    }

    private void loadDetailData() {
        final FutbolAPI futbolAPI = retrofit.create(FutbolAPI.class);
        Call<List<FutbolModel>> call = futbolAPI.getData();
        call.enqueue(new Callback<List<FutbolModel>>() {
            @Override
            public void onResponse(Call<List<FutbolModel>> call, Response<List<FutbolModel>> response) {
                if(response.isSuccessful()){
                    List<FutbolModel> responseDetailList = response.body();
                    futbolModels = new ArrayList<>(responseDetailList);

                    //RecyclerView
                    recyclerView.setLayoutManager(new LinearLayoutManager(DetailActivity.this));
                    recyclerViewAdapter = new DetailAdapter(futbolModels);
                    recyclerView.setAdapter(recyclerViewAdapter);

                    date.setText(futbolModels.get(0).date);

                }
            }

            @Override
            public void onFailure(Call<List<FutbolModel>> call, Throwable t) {
            t.printStackTrace();
            }
        });
    }
}

File: app/src/main/java/com/edanuryildirim/futbolapi/view/DetailActivity.kt
package com.edanuryildirim.futbolapi.view
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.edanuryildirim.futbolapi.R
import com.edanuryildirim.futbolapi.adapter.DetailAdapter
import com.edanuryildirim.futbolapi.adapter.RecyclerviewAdapter
import com.edanuryildirim.futbolapi.model.FutbolModel
import com.edanuryildirim.futbolapi.repository.AppRepository
import com.edanuryildirim.futbolapi.util.Resource
import com.edanuryildirim.futbolapi.util.errorSnack
import com.edanuryildirim.futbolapi.viewModel.PicsViewModel
import com.edanuryildirim.futbolapi.viewModel.ViewModelProviderFactory
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_detail.*

class DetailActivity : AppCompatActivity() {

    private lateinit var viewModel: PicsViewModel
    lateinit var picsAdapter: DetailAdapter
    private var futbolModel:List<FutbolModel>?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)
        init()
    }

    private fun init() {
        recyclerViewDetail.setHasFixedSize(true)
        recyclerViewDetail.layoutManager = LinearLayoutManager(this)
        setupViewModel()
    }

    private fun setupViewModel() {
        val repository = AppRepository()
        val factory = ViewModelProviderFactory(application, repository)
        viewModel = ViewModelProvider(this, factory).get(PicsViewModel::class.java)
        getPictures()
    }

    private fun getPictures() {
        viewModel.picsData.observe(this, Observer { response ->
            when (response) {
                is Resource.Success -> {
                    hideProgressBar()
                    response.data?.let { picsResponse ->
                        picsAdapter = DetailAdapter(picsResponse)
                        futbolModel=picsResponse
                        recyclerViewDetail.adapter = picsAdapter
                    }
                }

                is Resource.Error -> {
                    hideProgressBar()
                    response.message?.let { message ->
                        rootLayoutt.errorSnack(message,Snackbar.LENGTH_LONG)
                    }

                }

                is Resource.Loading -> {
                    showProgressBar()
                }
            }
        })
    }

    private fun hideProgressBar() {
        progres.visibility = View.GONE
    }

    private fun showProgressBar() {
        progres.visibility = View.VISIBLE
    }

}